src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const expensesService = require('../services/expenses.services');
+const usersService = require('../services/users.services');
+
+const getAll = (req, res) => {
+  const userId = Number(req.query.userId);
+  const {
+    from,
+    to,
+    categories,
+  } = req.query;
+
+  let expenses = expensesService.getAll();
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === userId);
+  }
+
+  if (from) {
+    expenses = expenses.filter(({ spentAt }) => spentAt >= from);
+  }
+
+  if (to) {
+    expenses = expenses.filter(({ spentAt }) => spentAt <= to);
+  }
+
+  if (categories) {
+    expenses = expenses.filter(expense =>
+      categories.includes(expense.category)
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

Your code will crash if you have only 1 category